File: frontend/src/services/categoryService.ts
import { apiClient } from '@/config/apiClient';
import type { 
  ApiResponse, 
  CategoryResponse, 
  CategoryRequest 
} from '@/types';

export const categoryService = {
  getAll: (search?: string): Promise<ApiResponse<CategoryResponse[]>> => {
    return apiClient.get('/categories', { params: { search } });
  },

  getById: (id: number): Promise<ApiResponse<CategoryResponse>> => {
    return apiClient.get(`/categories/${id}`);
  },

  create: (data: CategoryRequest): Promise<ApiResponse<CategoryResponse>> => {
    return apiClient.post('/categories', data);
  },

  update: (id: number, data: CategoryRequest): Promise<ApiResponse<CategoryResponse>> => {
    return apiClient.put(`/categories/${id}`, data);
  },

  delete: (id: number): Promise<ApiResponse<void>> => {
    return apiClient.delete(`/categories/${id}`);
  },
};

File: backend/src/main/java/com/scim/service/CategoryService.java
package com.scim.service;

import com.scim.dto.request.CategoryRequest;
import com.scim.dto.response.CategoryResponse;

import java.util.List;

public interface CategoryService {

    List<CategoryResponse> getAll(String search);

    CategoryResponse getById(Long id);

    CategoryResponse create(CategoryRequest request);

    CategoryResponse update(Long id, CategoryRequest request);

    void delete(Long id);
}
